lms/djangoapps/courseware/views/views.py

```diff
@@ -1343,10 +1341,19 @@ def generate_user_cert(request, course_id):
         return HttpResponseBadRequest(_("Course is not valid"))
 
     if not is_course_passed(student, course):
+        log.info(u"User %s has not passed the course: %s", student.username, course_id)
         return HttpResponseBadRequest(_("Your certificate will be available when you pass the course."))
 
     certificate_status = certs_api.certificate_downloadable_status(student, course.id)
 
+    log.info(
+        u"User %s has requested for certificate in %s, current status: is_downloadable: %s, is_generating: %s",
+        student.username,
+        course_id,
+        certificate_status["is_downloadable"],
+        certificate_status["is_generating"],
```

Comment:
Do we need to make sure that properties `is_downloadable` and `is_generating` are always there in `certificate_status `?. I think it is better to use `.get`.

Reply:
`certificate_status` will always have these keys. 
Check [certificate_downloadable_status](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/certificates/api.py#L239-L239) for more details.